File: backend/internal/router/notification.go
package router

import (
	"github.com/gin-gonic/gin"

	"github.com/techblog/community/internal/middleware"
)

// registerNotificationRoutes 通知路由
func registerNotificationRoutes(g *gin.RouterGroup, deps *Dependencies) {
	notifications := g.Group("/notifications", middleware.Auth(deps.Cfg.JWTSecret))
	notifications.GET("", deps.NotifH.List)
	notifications.PUT("/:id", deps.NotifH.MarkRead)

	// 独立前缀避免与 /notifications/:id 冲突
	action := g.Group("/notification", middleware.Auth(deps.Cfg.JWTSecret))
	action.GET("/unread-count", deps.NotifH.UnreadCount)
	action.PUT("/read-all", deps.NotifH.MarkAllRead)
}

File: frontend/src/api/notification.ts
import { get, put } from '@/utils/request'
import type { Notification, PageResult } from '@/types'

export function listNotifications(params: { page?: number; page_size?: number; unread_only?: boolean }) {
  return get<PageResult<Notification>>('/notifications', params)
}

export function getUnreadCount() {
  return get<{ unread_count: number }>('/notification/unread-count')
}

export function markAllRead() {
  return put<Record<string, never>>('/notification/read-all')
}

export function markRead(id: number) {
  return put<{ id: number }>(`/notifications/${id}`)
}
